src/test/java/org/broadinstitute/barclay/argparser/CollectionArgumentUnitTests.java

```diff
@@ -277,6 +277,17 @@ public void testCollectionThatCannotBeAutoInitialized() {
         new CommandLineArgumentParser(o);
     }
 
+    class ImmutableCollectionArguments {
+        @Argument
+        public List<String> LIST = Collections.emptyList();
+    }
+
+    @Test(expectedExceptions = CommandLineException.CommandLineParserInternalException.class)
+    public void testCollectionThatIsImmmutable() {
+        final ImmutableCollectionArguments o = new ImmutableCollectionArguments();
+        new CommandLineArgumentParser(o).parseArguments(System.err, new String[]{"--LIST", "A"});
+    }
+
```

Comment:
We need to have tests for all combinations of APPEND_TO_COLLECTIONS, "null", and real values.

Reply:
Done. Added data provider and different combinations.